### ISSUE TITLE
Rest API Update Site do not update siteName

### ISSUE THREAD

**Issue by erickgonzalez** [dotCMS/core#20688]
Found when testing: #20638

Update Site does not update siteName

STR:
1- Create a new Site, e.g called `new.site.com`
2- Run the following REST API call to update the site and change the siteName.
```
PUT {{serverURL}}/api/v1/site?id={{siteId}}

{
    "siteName":"postman.host.com",
}
```
3- Throws a 404 error.

You should be able to update siteName without issues.

**Comment by bryanboza** [comment]
Fixed, tested on release-21.09 // Postgres // FF

**Comment by wezell** [comment]
PUT {{serverURL}}/api/v1/site?id={{siteId}}

{
    "siteName":"postman.host.com",
    "acceptDangerousConsequences": true
}
